lib/src/edhoc.rs

```diff
@@ -615,12 +625,15 @@ fn encode_message_1(
     output.len = 3 + raw_suites_len + P256_ELEM_LEN + 1;
 
     if let Some(ead_1) = ead_1 {
-        let ead_1 = encode_ead_item(ead_1);
-        output.content[output.len..output.len + ead_1.len].copy_from_slice(ead_1.as_slice());
-        output.len += ead_1.len;
+        let ead_1 = encode_ead_item(ead_1)?;
+        if output.extend_from_slice(ead_1.as_slice()).is_ok() {
+            Ok(output)
+        } else {
+            Err(EDHOCError::EncodingError)
+        }
```

Comment:
Did a similar change based on yours, I just can't return it like this because extend_from_slice mutates `output` instead of returning it.

Reply:
Right, because it's not returning a suitable self.

What works would be

```rust
        plaintext_3.extend_from_slice(ead_3.as_slice())
            .and(Ok(plaintext_3))
            .or(Err(EDHOCError::EadTooLongError))
```

or

```rust
        plaintext_3.extend_from_slice(ead_3.as_slice())
            .map(|()| plaintext_3)
            .map_err(|()| EDHOCError::EadTooLongError)
```

but at that point, readability is not really improved a lot any more.

The prettiest I can come up with that works is

```rust
    if let Some(ead_3) = ead_3 {
        let ead_3 = encode_ead_item(ead_3)?;
        plaintext_3.extend_from_slice(ead_3.as_slice())
            .or(Err(EDHOCError::EadTooLongError))?;
    }
    Ok(plaintext_3)
```

but again I don't know how often we want to take that path here.

Reply:
Adopted the first suggestion. Also, used a match to avoid the early return on `encode_ead_item`.